### PR TITLE
Support LayerNorm for FC and Normalizer as layer

### PR DESCRIPTION
**Pull request by emailweixu**

### CODE REVIEW

File: alf/layers.py [hunk 1]
```diff
@@ -309,6 +328,10 @@ def forward(self, inputs):
             y = inputs.matmul(self._weight.t())
             if self._use_bias:
                 y += self._bias
+        if self._use_ln:
```
Comment (hnyu) [inline alf/layers.py:331]: What if self._use_ln and self._use_bn are both True? Why it's the current order being applied?

File: alf/utils/normalizers.py [hunk 2]
```diff
@@ -171,6 +173,12 @@ def _normalize(m2, t, m=None):
             return alf.nest.map_structure(_normalize, self._m2_averager.get(),
                                           tensor)
 
+    def forward(self, input):
+        if self.training:
```
Comment (hnyu) [inline alf/utils/normalizers.py:177]: Will this self.training flag automatically be set?

**Review by hnyu — CHANGES_REQUESTED** [review]
nn.LayerNorm is still different from ALF Normalizer because it uses mean and std from the current input batch, instead of historical data.

File: alf/layers.py (hunk 1 above)
Comment (emailweixu) [reply alf/layers.py:331]: In general, they should not be applied at the same time.

File: alf/utils/normalizers.py (hunk 2 above)
Comment (emailweixu) [reply alf/utils/normalizers.py:177]: Yes.

**Review by emailweixu — COMMENTED** [review]
Right.

> nn.LayerNorm is still different from ALF Normalizer because it uses mean and std from the current input batch, instead of historical data.

File: alf/layers.py (hunk 1 above)
Comment (hnyu) [reply alf/layers.py:331]: So maybe add a check for exclusion?

Comment (emailweixu) [reply alf/layers.py:331]: But I guess if someone really want to use both, that's ok.

**Review by hnyu — APPROVED** [review]
